task/net-api/src/lib.rs

```diff
@@ -49,6 +50,116 @@ pub enum PhyError {
     Other = 3,
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq, FromPrimitive, IdolError)]
+#[repr(u32)]
+pub enum KszError {
+    /// This functionality is not available on the given board
+    NotAvailable,
+    /// The MAC table index is too large
+    BadMacIndex,
+    /// The given address is not a valid register
+    BadRegister,
+
+    WrongChipId,
+
+    // Errors copied from SpiError
+    BadTransferSize,
+    ServerRestarted,
+    NothingToRelease,
+    BadDevice,
+    DataOverrun,
+}
+
+#[cfg(feature = "ksz8463")]
+impl From<ksz8463::Error> for KszError {
+    fn from(e: ksz8463::Error) -> Self {
+        use drv_spi_api::SpiError;
+        match e {
+            ksz8463::Error::SpiError(e) => match e {
+                SpiError::BadTransferSize => KszError::BadTransferSize,
+                SpiError::ServerRestarted => KszError::ServerRestarted,
+                SpiError::NothingToRelease => KszError::NothingToRelease,
+                SpiError::BadDevice => KszError::BadDevice,
+                SpiError::DataOverrun => KszError::DataOverrun,
+            },
+            ksz8463::Error::WrongChipId(..) => KszError::WrongChipId,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, AsBytes, FromBytes)]
+#[repr(C)]
+pub struct KszMacTableEntry {
+    pub mac: [u8; 6],
+    pub port: u16,
+}
+
+#[cfg(feature = "ksz8463")]
+impl From<ksz8463::KszRawMacTableEntry> for KszMacTableEntry {
+    fn from(e: ksz8463::KszRawMacTableEntry) -> Self {
+        Self {
+            mac: e.addr,
+            port: match e.source {
+                ksz8463::SourcePort::Port1 => 1,
+                ksz8463::SourcePort::Port2 => 2,
+                ksz8463::SourcePort::Port3 => 3,
+            },
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, AsBytes, FromBytes)]
+#[repr(C)]
+pub struct MacAddress(pub [u8; 6]);
+
+#[derive(Copy, Clone, Debug, Default, Serialize, Deserialize)]
+#[repr(C)]
+pub struct ManagementLinkStatus {
+    pub ksz8463_100base_fx_link_up: [bool; 2],
+    pub vsc85x2_100base_fx_link_up: [bool; 2],
+    pub vsc85x2_sgmii_link_up: [bool; 2],
+}
+
+#[derive(Copy, Clone, Debug, Default, Serialize, Deserialize)]
+#[repr(C)]
+pub struct ManagementCountersVsc85x2 {
+    pub mac_good: u16,
+    pub media_good: u16,
+    pub mac_bad: u16,
+    pub media_bad: u16,
+}
+
+#[derive(Copy, Clone, Debug, Default, Serialize, Deserialize)]
+#[repr(C)]
+pub struct ManagementCountersKsz8463 {
+    pub multicast: u32,
+    pub unicast: u32,
+    pub broadcast: u32,
+}
+
+#[derive(Copy, Clone, Debug, Default, Serialize, Deserialize)]
+#[repr(C)]
+pub struct ManagementCounters {
+    pub vsc85x2_tx: [ManagementCountersVsc85x2; 2],
+    pub vsc85x2_rx: [ManagementCountersVsc85x2; 2],
+
+    pub ksz8463_tx: [ManagementCountersKsz8463; 3],
+    pub ksz8463_rx: [ManagementCountersKsz8463; 3],
+
+    /// The MAC counters are only valid on the VSC8562
+    pub vsc85x2_mac_valid: bool,
```

Comment:
I don't see this being set anywhere; is it always false?

Reply:
Good catch, I forgot to set it!

(That's the downside of using `Default` then filling in fields later, although I'm not sure if there's a better way to initialize this struct...)